edx_adapt/api/resources/course_resources.py

```diff
@@ -12,78 +13,56 @@
 course_parser.add_argument('course_id', type=str, required=True, location='json', help="Please supply a course ID")
 
 
-class Courses(Resource):
-    def __init__(self, **kwargs):
-        self.repo = kwargs['data']
-        """@type repo: DataInterface"""
-
-    def get(self):
-        courses = []
+class BaseResources(BaseResource):
+    def _post_request(self, function_name, *args, **kwargs):
         try:
-            courses = self.repo.get_course_ids()
+            getattr(self.repo, function_name)(*args, **kwargs)
         except DataException as e:
```

Comment:
Here we using getattr for interface's methods which we are responsible for that is why Attribute errors are not expected.